### ISSUE TITLE
Extra spaces added by GFM special comment parsing cause verbatim blocks

### ISSUE THREAD

**Issue by SRGDamia1** [doxygen/doxygen#11071]
**Describe the bug**

In parsing special markdown comments into Doxygen commands, extra spaces are added to the output:

https://github.com/doxygen/doxygen/blob/c692605decbd3f78054dfc780da6749235883cb6/src/commentcnv.l#L616-L620

These extra spaces cause the downstream markdown parser to interpret the commentcnv output as a verbatim block code block rather than as a command.

These features were added in #10959 

**Screenshots**
 
In this markdown:

```md
<!--! @page test_page A Test Page -->
# A Test Page - Page to Test Markdown Parser

Test TOC command

<!--! @tableofcontents -->

Test warning command

<!--! @warning Will this warn me? -->
```

The special commands are rendered as verbatim blocks rather than being interpreted as a command:

![image](https://github.com/user-attachments/assets/480ac64d-fbfb-4352-a2ab-e71f0794b22b)

**To Reproduce**

I used only the markdown above and the default doxyfile.

**Expected behavior**

The command should be invisible in rendered GItHub markdown, but seen as a command by Doxygen

**Version**

1.12.0 (c73f5d30f9e8b1df5ba15a1d064ff2067cbb8267)
    with sqlite3 3.42.0 and clang support 13.0.0.

**Additional context**

Looking at the CommentCnv and Markdown debugging

The original markdown: 

```md
<!--! @page test_page A Test Page -->
# A Test Page - Page to Test Markdown Parser

Test TOC command

<!--! @tableofcontents -->

Test warning command

<!--! @warning Will this warn me? -->

```

After CommentCnv processing, which becomes the input to the markdown processor: [Note the extra spaces before each command!]

```md
      @page test_page A Test Page 
# A Test Page - Page to Test Markdown Parser

Test TOC command

      @tableofcontents 

Test warning command

      @warning Will this warn me? 
```

The markdown processor output then gives: [Note the @iverbatim block!]

```dox
      @page md_test__file A Test Page \ilinebr @ianchor{A Test Page} test_page
Page 
@section autotoc_md0 A Test Page - Page to Test Markdown Parser

Test TOC command
@iverbatim
  @tableofcontents 
@endiverbatim\ilinebr 
Test warning command
@iverbatim
  @warning Will this warn me? 
@endiverbatim\ilinebr 
```

**Comment by albert-github** [comment]
This issue reveals 2 more problems:
- documentation, see proposed pull request #11072
- spaces before the `@page` command result in repeating the last part of the `@page` title,  see proposed pull request #11073
- the problem in this issue: I've just pushed a proposed patch, pull request #11074

(I see the problems as 3 separate problems, hence 3 PR)

**Comment by SRGDamia1** [comment]
Thank you for the quick response!

**Comment by albert-github** [comment]
- Code of #11072 (documentation issue) has been integrated in master on GitHub

**Comment by albert-github** [comment]
- Code of #11074 (the original problem) has been integrated in master on GitHub

**Comment by albert-github** [comment]
Code of issue #11073 has been integrated in master on GitHub.
Now all subparts are integrated into the master version.
(please don't close the issue as this will be done at the moment of an official release).

**Comment by SRGDamia1** [comment]
Thank you!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.13.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
